### PR TITLE
Afform - Support multiple permissions in the GUI

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
Allow multiple permissions to be set for an Afform.

Before
----------------------------------------
You can do it in code by editing the `.aff.json` file, and it pretty much works. Pass an array for AND. A nested array works for OR for the page route but causes errors for a navigation menu item.
But none of that works in the GUI.

After
----------------------------------------
Both the GUI and the API now support the same data structure: `permission` is a (non-nested) array, and `permission_operator` can be AND or OR. This is exactly the same structure as `civicrm_navigation`.

image

### CODE REVIEW

File: ext/afform/core/afform.php [hunk 1]
```diff
@@ -443,14 +443,21 @@ function afform_civicrm_permission_check($permission, &$granted, $contactId) {
...
-    $afform = \Civi\Api4\Afform::get()
-      ->setCheckPermissions(FALSE)
+    $afform = \Civi\Api4\Afform::get(FALSE)
       ->addWhere('name', '=', $name)
-      ->setSelect(['permission'])
+      ->addSelect('permission', 'permission_operator')
       ->execute()
       ->first();
+    // No permissions found... this shouldn't happen but just in case, set default.
+    if ($afform && empty($afform['permission'])) {
```
[the remark this answers is not in the document]
Comment (kurund) [reply ext/afform/core/afform.php:452]: Also permission is not required in the UI.

**Review by kurund — COMMENTED** [review]
@colemanw 

Everything else looks good to me and working as expected.

File: ext/afform/core/afform.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (colemanw) [reply ext/afform/core/afform.php:452]: @kurund I didn't want to change the status-quo in this PR. Currently `AfformScanner` forces it to be `'access CiviCRM'` if not set:
https://github.com/civicrm/civicrm-core/blob/83af1505cabe631582a1f29466052ee4a7bd9747/ext/afform/core/CRM/Afform/AfformScanner.php#L152-L161

[the remark this answers is not in the document]
Comment (kurund) [reply ext/afform/core/afform.php:452]: Oh ok that's fine then.

Currently, we can set.

```bash
"permission": [],
```
that will skip permissions.

[the remark this answers is not in the document]
Comment (colemanw) [reply ext/afform/core/afform.php:452]: @kurund I think the way to skip permissions is to set it to `['*always allow*']`

**Pull request merged**